core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java

```diff
@@ -242,10 +242,10 @@ private URI endpointFromConfig(SdkClientConfiguration config) {
             .withRegion(config.option(AwsClientOption.AWS_REGION))
             .withProfileFile(() -> config.option(SdkClientOption.PROFILE_FILE))
             .withProfileName(config.option(SdkClientOption.PROFILE_NAME))
-            .putAdvancedOption(ServiceMetadataAdvancedOption.DEFAULT_S3_US_EAST_1_REGIONAL_ENDPOINT,
-                               config.option(ServiceMetadataAdvancedOption.DEFAULT_S3_US_EAST_1_REGIONAL_ENDPOINT))
-            .withDualstackEnabled(config.option(AwsClientOption.DUALSTACK_ENDPOINT_ENABLED))
-            .withFipsEnabled(config.option(AwsClientOption.FIPS_ENDPOINT_ENABLED))
+            // .putAdvancedOption(ServiceMetadataAdvancedOption.DEFAULT_S3_US_EAST_1_REGIONAL_ENDPOINT,
```

Comment:
Why is this commented  ? Can we remove this or Add a TODO: to remove this

Reply:
This is intentional, needed to make some of tests pass in `endpoint-tests.json` that test behavior that conflicts with this; for example they do their own validation when setting both a custom endpoint and FIPS/dual stack. I'll add note.